tempodb/search/rescan_blocks.go

```diff
@@ -27,41 +26,48 @@ func RescanBlocks(walPath string) ([]*StreamingSearchBlock, error) {
 
 	blocks := make([]*StreamingSearchBlock, 0, len(files))
 	for _, f := range files {
-		if f.IsDir() {
+		info, err := f.Info()
```

Comment:
IsDir is no longer allowed?

Reply:
switching from ioutil.ReadDir to os.ReadDir, it is no longer returning FileInfo. 

```
info, err := f.Info()
if info.IsDir() {
``` 
is the workaround